packages/presentation/presentation_containers/interfaces/publishes_settings.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { PublishingSubject } from '@kbn/presentation-publishing';
+
+export interface PublishesSettings {
+  settings: Record<string, PublishingSubject<boolean | undefined>>;
```

Comment:
This is a really powerful way of setting up our settings provider. We might eventually need some more type guards that can make fetching the individual settings safer, but it's a great start.